### PR TITLE
Add a large tutorial and integrate Literate.jl

### PR DESCRIPTION
**Pull request by slwu89**
Hi @thevolatilebit, this PR does 2 main things:

  1. Adds a large tutorial in tutorials/stochastic_simulation/anderson.jl demonstrating how to use AlgebraicAgents.jl to implement the stochastic simulation algorithm of [Anderson (2007)](https://people.math.wisc.edu/~dfanderson/papers/AndNRM.pdf), hopefully it will make more clear the concepts of the package to users, and demonstrate the generality of the software.
  2. Use [Literate.jl](https://github.com/fredrikekre/Literate.jl) to build the file tutorials/stochastic_simulation/anderson.jl into a markdown document which can be parsed by Documenter.jl, this involves adding Literate.jl as a dependency to docs/Project.toml and adding a line which calls `Literate.markdown` to docs/make.jl. This can serve as an example of how to migrate the other tutorials to documents which can be compiled by Literate.jl into Documenter.jl-friendly markdown, and in turn become part of the generated website.

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -28,6 +28,16 @@ design_mmd = init_mermaid * design
 open(joinpath(dirname(@__FILE__), "src/design_mmd.md"), "w") do io
     write(io, design_mmd)
 end
+# end required for mmd
+
+# Literate for tutorials
+
+Literate.markdown(
```
Comment (thevolatilebit) [inline docs/make.jl:35]: I wonder if we want to keep the tutorial generated by Literate in /docs? I see that Agents.jl take a similar approach, but their repo does not contain the generated files (they do not delete them either, just run the build step outside)

Comment (thevolatilebit) [reply docs/make.jl:35]: No effect except for simpler diffs.

Comment (slwu89) [reply docs/make.jl:35]: Thanks @thevolatilebit, happy to do whatever changes to keep the repo cleaner! Do you mean just writing the ouput to src/anderson.md? I am looking at this line of Agents.jl https://github.com/JuliaDynamics/Agents.jl/blob/main/docs/make.jl#L29

Comment (slwu89) [reply docs/make.jl:35]: I cannot understand how Agents.jl keeps the repo so clean, it looks like they do not do any cleaning up within the make.jl file after making all those .md files using `Literate.markdown`. It does seem like they put them into a folder that is included in .gitignore (https://github.com/JuliaDynamics/Agents.jl/blob/main/.gitignore), and the file directly added to src/ is also in the .gitignore (agents_visualizations.md).

Comment (slwu89) [reply docs/make.jl:35]: Anyways, I defer to your suggestion on how we should keep the repo cleaner, either by deleting generated files at the end of make.jl or putting them into a folder that is gitignore-d.

Comment (thevolatilebit) [reply docs/make.jl:35]: Yes, it definitely  does not matter now. But in parallel with removing `design_mmd.md`, we could delete the generated file using something like

```julia
rm(joinpath(@__DIR__, "src/sketches/stochastic_simulation/anderson.md"))
```

Comment (slwu89) [reply docs/make.jl:35]: Thanks!

**Review by thevolatilebit — APPROVED** [review]
I really appreciate the Anderson example!

**Pull request merged**